### PR TITLE
Add the Cluster Autoscaler addon

### LINKED ISSUE #391: Integrate cluster-autoscaler as addon
based on https://github.com/kubernetes/autoscaler/pull/1866

### PR DESCRIPTION
**Pull request by xmudrii**
**What this PR does / why we need it**:

This PR adds the Cluster Autoscaler addon, including documentation on how to use it. The documentation will be added to the docs website once this PR is merged.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #391 

**Does this PR introduce a user-facing change?**:
```release-note
Add the Cluster Autoscaler addon
```

/assign @kron4eg

### CODE REVIEW

File: addons/cluster-autoscaler/README.md [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+The Cluster Autoscaler only considers MachineDeployment with the valid 
+annotations. The annotations are used to control the minimum and maximum number
+of replicas per MachineDeployment:
+
+* `cluster.k8s.io/cluster-api-autoscaler-node-group-min-size` - the minimum
+  number of replicas (must be greater than zero)
+* `cluster.k8s.io/cluster-api-autoscaler-node-group-max-size` - the maximum
+  number of replicas
+
+**Note:** You don't need to apply those labels to all MachineDeployment
```
Comment (kron4eg) [inline addons/cluster-autoscaler/README.md:77]:
```suggestion
**Note:** You don't need to apply those annotations to all MachineDeployment
```

**Review by kron4eg — COMMENTED** [review]
awesome 🎉 !

(micro nit in comments)

File: addons/cluster-autoscaler/README.md (hunk 1 above)
Comment (xmudrii) [reply addons/cluster-autoscaler/README.md]: @kron4eg Fixed, PTAL.

**Review by kron4eg — APPROVED** [review]

**Pull request merged**